User name with spaces doesnt work

When I try to create a user with spaces it crashes. You can easily simulate this by putting a space into username in the test `TestAccBitbucketUser` like this `name = "admin %v"`

```
=== RUN   TestAccBitbucketUser
    TestAccBitbucketUser: testing.go:568: Step 0 error: errors during apply:
        
        Error: API Error: 404 /rest/api/1.0/users/admin+7292545641539942086 User admin+7292545641539942086 does not exist.
        
        
          on /var/folders/t6/mhc7m08j29d0jvfzxc2_zzr40000gn/T/tf-test963642486/main.tf line 2:
          (source code not available)
        
        
    TestAccBitbucketUser: testing.go:629: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during refresh: failed to get user admin 7292545641539942086 from bitbucket: API Error: 404 /rest/api/1.0/users/admin+7292545641539942086 User admin+7292545641539942086 does not exist.
        
        
        State: <nil>
--- FAIL: TestAccBitbucketUser (0.20s)
```

--- a/bitbucket/resource_user.go
+++ b/bitbucket/resource_user.go
@@ -152,7 +152,7 @@ func resourceUserRead(d *schema.ResourceData, m interface{}) error {
 
 	client := m.(*BitbucketServerProvider).BitbucketClient
 	req, err := client.Get(fmt.Sprintf("/rest/api/1.0/users/%s",
-		url.QueryEscape(name),
+		url.PathEscape(name),
 	))
 
 	if err != nil {
@@ -192,7 +192,7 @@ func resourceUserExists(d *schema.ResourceData, m interface{}) (bool, error) {
 
 	client := m.(*BitbucketServerProvider).BitbucketClient
 	req, err := client.Get(fmt.Sprintf("/rest/api/1.0/users/%s",
-		url.QueryEscape(name),
+		url.PathEscape(name),
 	))
 
 	if err != nil {

--- a/bitbucket/resource_user_test.go
+++ b/bitbucket/resource_user_test.go
@@ -15,7 +15,7 @@ func TestAccBitbucketUser(t *testing.T) {
 	userRand := fmt.Sprintf("%v", rand.New(rand.NewSource(time.Now().UnixNano())).Int())
 	config := fmt.Sprintf(`
 		resource "bitbucketserver_user" "test" {
-			name = "admin%v"
+			name = "admin %v"
 			display_name = "Admin %v"
 			email_address = "admin%v@example.com"
 		}
@@ -31,7 +31,7 @@ func TestAccBitbucketUser(t *testing.T) {
 			{
 				Config: config,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("bitbucketserver_user.test", "name", "admin"+userRand),
+					resource.TestCheckResourceAttr("bitbucketserver_user.test", "name", "admin "+userRand),
 					resource.TestCheckResourceAttr("bitbucketserver_user.test", "display_name", "Admin "+userRand),
 					resource.TestCheckResourceAttr("bitbucketserver_user.test", "email_address", "admin"+userRand+"@example.com"),
 					resource.TestCheckResourceAttrSet("bitbucketserver_user.test", "initial_password"),
@@ -40,7 +40,7 @@ func TestAccBitbucketUser(t *testing.T) {
 			{
 				Config: configModified,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("bitbucketserver_user.test", "name", "admin"+userRand),
+					resource.TestCheckResourceAttr("bitbucketserver_user.test", "name", "admin "+userRand),
 					resource.TestCheckResourceAttr("bitbucketserver_user.test", "display_name", "Admin Updated "+userRand),
 					resource.TestCheckResourceAttr("bitbucketserver_user.test", "email_address", "admin"+userRand+"@example.com"),
 					resource.TestCheckResourceAttrSet("bitbucketserver_user.test", "initial_password"),

